AdblockPlusSafariExtension/AdblockPlus+Extension.m

```diff
@@ -53,4 +54,28 @@ - (NSURL *)currentFilterListURL
   return [[NSBundle mainBundle] URLForResource:[filename stringByDeletingPathExtension] withExtension:@"json"];
 }
 
+- (NSURL *)activeFilterListURLWithWhitelistedWebsites
+{
+  NSURL *original = self.activeFilterListsURL;
+  NSString *filename = original.lastPathComponent;
+
+  if (filename == nil || [filename isEqual:@"empty.json"] || self.whitelistedWebsites.count == 0)  {
+    return original;
+  }
+
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+  NSURL *copy = [fileManager containerURLForSecurityApplicationGroupIdentifier:self.group];
+  copy = [copy URLByAppendingPathComponent:[NSString stringWithFormat:@"ww-%@", filename] isDirectory:NO];
```

Comment:
Copy with merged whitelisting rules is always created, before it served to content blocker. Actually, I do not know if it is good idea to store it in persistent memory, but I do not removing it after it is processed.